src/main/java/com/github/onsdigital/babbage/api/util/LegacyProxyMaxAge.java

```diff
@@ -0,0 +1,64 @@
+package com.github.onsdigital.babbage.api.util;
+
+import java.util.Enumeration;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.http.Header;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import com.github.onsdigital.logging.v2.event.SimpleEvent;
+import static com.github.onsdigital.babbage.configuration.ApplicationConfiguration.appConfig;
+
+public class LegacyProxyMaxAge {
+    public int getMaxAgeValueFromLegacyProxy(HttpServletRequest request) {
+        RequestBuilder requestBuilder = buildRequest(request);          
+        HttpUriRequest forwardRequest = requestBuilder.build();
+
+        int maxAgeValue = 0;
+        try {
+            CloseableHttpClient httpClient = HttpClients.createDefault();
+            CloseableHttpResponse proxyResponse = httpClient.execute(forwardRequest);
+            maxAgeValue = extractMaxAgeValue(proxyResponse);
+        } catch (Exception e) {
+            SimpleEvent.error().exception(e).log("api MaxAge: error getting max age from legacy proxy");
+        }
+        return maxAgeValue;    
+    }
+    
+    private RequestBuilder buildRequest(HttpServletRequest request) {
+        RequestBuilder requestBuilder = RequestBuilder.create(request.getMethod())
+                .setUri(appConfig().babbage().getLegacyCacheProxyUrl());
+    
+        Enumeration<String> headerNames = request.getHeaderNames();
```

Comment:
Are you sure all this is needed? Now that I'm reading the original `getMaxAge::getMaxAge` method more carefully, it looks like all you need from the original request is the `uri` parameter. That will be just a string with the URI for the resource. If you check our 3rd diagram, you'll see that this corresponds with a "resource endpoint", so I think that you can potentially simplify the `getMaxAgeValueFromLegacyProxy` method and require a `String resourceURI` instead of a `request`. But please don't take my word for it, verify that this makes sense, I could be wrong.

Reply:
Yes, you are correct about this, we only need to copy the URI parameter from the original request!